src/transfer.js

```diff
@@ -0,0 +1,346 @@
+import { getFileHash } from './utils';
+import axios from 'axios';
+/**
+ * @typedef {Object} BasicLfsAction An LFS basic transfer mode action descriptor
+ * @property {String} href
+ * @property {Object<String, String>} header
+ * @property {Number} expires_in
+ */
+
+class GitLfsBasicTransfer {
+
+  /**
+   * Create a new 'basic' transfer adapter
+   *
+   * @param {Object} actions
+   * @param {BasicLfsAction} actions.upload
+   * @param {BasicLfsAction} actions.verify
+   * @param {File} file
+   * @param {CallableFunction} onProgress
+   */
+  constructor(actions, file, onProgress) {
+    this._file = file;
+    this._onProgress = onProgress;
+
+    this._uploadAction = actions.upload || null;
+    this._verifyAction = actions.verify || null;
+  }
+
+  /**
+   * Upload the file to storage, using the protocol specified by the
+   * transfer adapter
+   *
+   * @returns {Promise<boolean>}
+   */
+  async upload() {
+    let response;
+    if (this._uploadAction) {
+      response = await this._doUpload();
+      if (! (response.status >= 200 && response.status < 300)) {
+        throw `'upload' action failed with HTTP ${response.status}`;
+      }
+    }
+
+    if (this._verifyAction) {
+      response = await this._doVerify();
+      if (response.status !== 200) {
+        if (response.message) {
+          throw response.message
+        } else {
+          throw `'verify' action failed with HTTP ${response.status}`;
+        }
+      }
+    }
+
+    return true;
+  }
+
+  async _doUpload() {
+    const { href, header } = this._uploadAction;
+    const body = this._file.descriptor;
+    const config = {
+      headers: {
+        "Content-type": this._file.descriptor.type || 'application/octet-stream',
+        ...header
+      },
+    }
+
+    if (this._onProgress) {
+      config.onUploadProgress = this._onProgress
+    }
+
+    return axios.put(href, body, config);
```

Comment:
you can destruct the Axios response before return if you will need more than just the data to be returned.

If 
e.g. : 
```
const { data, status } = await axios.put(href, body, config);

return { data, status } 
```